ledger/accountdb.go

```diff
@@ -1208,7 +1143,11 @@ func (iterator *orderedAccountsIter) Next(ctx context.Context) (acct []accountAd
 	}
 	if iterator.step == oaiStepCreateOrderingTable {
 		// create the temporary table
-		_, err = iterator.tx.ExecContext(ctx, "CREATE TABLE accountsiteratorhashes(address blob, hash blob)")
+		if iterator.fetchAccountData {
+			_, err = iterator.tx.ExecContext(ctx, "CREATE TABLE accountsiteratorhashes(accountbase_rawid INTEGER, hash blob)")
```

Comment:
Have you considered adding accountbase.data column to accountsiteratorhashes? 
It looks like the accountbase.data is being read once during the select in oaiStepInsertAccountData, and once during the join in oaiStepSelectFromOrderedTable. 

Copying them over to accountsiteratorhashes will avoid the join, as well as scanning all pages with data from accountbase, most of which will not be in the result anyway.

Reply:
Yes, I did. Unfortunately adding column means that we'll pay a higher price on deletion/insertion to the table.
I was trying to avoid doing that if possible.
regardless, I think that this PR is going to be shelved. If you could review https://github.com/algorand/go-algorand/pull/1712 PR instead, it would be great. ( it's a subset of this one )